components/vni.js

```diff
@@ -0,0 +1,63 @@
+// Vni.js
+
+var _ = require('underscore');
+var Lm = require('./Lm');
+var state = require('./state');
+
+// Hash of vnis
+vnis = []; 
+
+// Define Vni class and its subclasses - InterNodeVni & IipVni
+function Vni( vnid ) {
+  this.vnid = vnid;
+  this.lm = Lm();
```

Comment:
A VNI does not have an lm.  A State has an lm, but should be left undefined initially.